File: src/app/models/FairSolution.ts
import {DecisionNode} from "./DecisionNode";
import {AssessmentResult, IndicatorGroup} from "./FairAssessment";
import {FairResource} from "./FairResource";

export interface FairSolution {
  id: string;
  link: string;
  createDate: string;
  title: string;
  project: string;
  name: string;
  email: string;
  description: string;
  comments: string;

  decisionTree: DecisionNode[];
  labels: string[]; // this could be interested categories
  indicatorGroups: IndicatorGroup[];
  assessment: AssessmentResult;
  fairResources: FairResource[]; // could be filtered by decision tree results or/and assessment results
}

File: src/main/java/uk/ac/ebi/fairwizard/model/FairSolution.java
package uk.ac.ebi.fairwizard.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class FairSolution {
  public String id;
  public String link;
  public String createDate;
  public String title;
  public String project;
  public String description;
  public String comments;

  public List<DecisionNode> decisionTree;
  public Set<String> labels; // this could be interested categories
  public List<IndicatorGroup> indicatorGroups;
  public FairAssessment assessment;
  public Set<MongoFairResource> fairResources; // could be filtered by decision tree results or/and assessment results

  public FairSolution(String id) {
    this.id = id;
  }
}
